data-weave-plugin/src/main/java/org/mule/tooling/restsdk/utils/RestSdkHelper.java

```diff
@@ -41,14 +42,18 @@
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
 import java.util.stream.Stream;
 
 public class RestSdkHelper {
+  public static final Logger LOGGER = Logger.getInstance(RestSdkHelper.class);
 
   public static SelectionPath swaggerVersion = SelectionPath.DOCUMENT.child("swagger");
   public static SelectionPath openApiVersion = SelectionPath.DOCUMENT.child("openapi");
 
+  final private static ConcurrentHashMap<Object, Object> dummyMapForSynchronizingPerKey = new ConcurrentHashMap<>();
```

Comment:
This will produce a memory leak. Using PSIElements as key is not recommended.

Reply:
I'm not actually storing anything in the map. The lambda function used below always returns null. I'm using this map just for the locking, because while calling the lambda `ConcurrentHashMap.computeIfAbsent()` creates a key-specific lock and provides synchronization. It just uses the hashcode for this, it doesn't store the object. However, if you think it would be better I can pass a different thing, like the API url.

Reply:
@machaval  Would you prefer a simple _synchronized_ instead? I don't think there would be many concurrent parsing of different APIs anyway...

Reply:
@machaval  Hi! I'd like to merge a version of this. Without this the plugin crashes with large APIs. Thanks!